hips_viewer/src/map.ts

```
@@ -195,3 +201,58 @@ export function updateColors() {
     pointFeature.value.style('fillColor', styleCellFunction).draw()
   }
 }
+
+export function cellDistribution() {
+  if (!(colormapName.value && cells.value && histSelectedCells.value)) return []
+  const selectedCells = cells.value.filter((cell: any) => histSelectedCells.value.has(cell.id))
+
+  const values = [...new Set(cells.value.map(
+    (cell: any) => getCellAttribute(cell, colorBy.value),
+  ).map(
+    (v: any) => isNaN(parseFloat(v)) ? v : parseFloat(v),
+  ).filter((v: any) => v !== undefined)),
+  ].filter((v: any) => typeof v === 'number' || typeof v === 'string')
+
+  const counts: Record<string | number, number> = {}
+  selectedCells.forEach((cell: any) => {
+    const key = getCellAttribute(cell, colorBy.value)
+    if (key !== undefined) counts[key] = (counts[key] ?? 0) + 1
+  })
+
+  // @ts-ignore
+  const colormapSets = colorbrewer[colormapName.value]
+
+  const buckets = histNumBuckets.value
+  showHistogram.value = values.length > buckets
+
+  if (!showHistogram.value || !values.every(v => typeof v === 'number')) {
```

Why do we return this if `showHistogram.value` is false? I would expect that check at the top of this function with the other store vars that must exist (where we return `[]` if not).

---

Let me change the if statement.. the name of the variable is confusing. Basically, `showHistogram` controls whether we're showing data that's already categorical (where the # of values is < the # of buckets), or data that needs to be put into calculated histogram buckets. So when `showHistogram = false` we use the default color pallete and hide the `histNumBuckets` slider.

---

Oh I see, that makes sense. After your change, I don't think it's necessary to rename it. I'll resolve this convo